pkg/minikube/driver/driver.go

```diff
@@ -234,5 +234,13 @@ func MachineName(cc config.ClusterConfig, n config.Node) string {
 	if len(cc.Nodes) == 1 || n.ControlPlane {
 		return cc.Name
 	}
-	return fmt.Sprintf("%s-%s", cc.Name, n.Name)
+	return fmt.Sprintf("%s---%s", cc.Name, n.Name)
+}
+
+// ClusterNameFromMachine retrieves the cluster name embedded in the machine name
```

Comment:
I see pain and suffering on this path. Instead of adding structure to an unstructured field & calling config.Load outside of main, we should be passing the config object around.